public/app/plugins/datasource/logging/language_provider.ts

```diff
@@ -158,6 +157,38 @@ export default class LoggingLanguageProvider extends LanguageProvider {
     return { context, refresher, suggestions };
   }
 
+  async importPrometheusQuery(query: string): Promise<string> {
```

Comment:
this could need a unit test. or do you think it might radically change and it is just a poc that will be thrown away?

Reply:
Yup, will add unit test!